compendium/DeclarativeServices/test/SharedLibraryExceptionTest.cpp

```diff
@@ -0,0 +1,111 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=============================================================================*/
+
+#include <gtest/gtest.h>
+
+#include <cppmicroservices/Bundle.h>
+#include <cppmicroservices/BundleEvent.h>
+#include <cppmicroservices/BundleContext.h>
+#include "cppmicroservices/Framework.h"
+#include "cppmicroservices/FrameworkEvent.h"
+#include "cppmicroservices/FrameworkFactory.h"
+#include "cppmicroservices/SharedLibraryException.h"
+
+#include "cppmicroservices/servicecomponent/runtime/ServiceComponentRuntime.hpp"
+
+#include "TestInterfaces/Interfaces.hpp"
+#include "TestUtils.hpp"
+
+namespace scr = cppmicroservices::service::component::runtime;
+
+class SharedLibraryExceptionTest : public ::testing::Test
+{
+protected:
+  SharedLibraryExceptionTest()
+    : framework(cppmicroservices::FrameworkFactory().NewFramework())
+  { }
+      
+  virtual ~SharedLibraryExceptionTest() = default;
+
+  virtual void SetUp()
+  {
+    framework.Start();
+    auto context = framework.GetBundleContext();
+
+    auto dsPluginPath = test::GetDSRuntimePluginFilePath();
+    auto dsbundles = context.InstallBundles(dsPluginPath);
+    for (auto& bundle : dsbundles) {
+      bundle.Start();
+    }
+    
+    auto sRef = context.GetServiceReference<scr::ServiceComponentRuntime>();
+    ASSERT_TRUE(sRef);
+    dsRuntimeService = context.GetService<scr::ServiceComponentRuntime>(sRef);
+    ASSERT_TRUE(dsRuntimeService);
+  }
+
+  virtual void TearDown()
+  {
+    framework.Stop();
+    framework.WaitForStop(std::chrono::milliseconds::zero());
+  }
+
+  cppmicroservices::Framework& GetFramework() { return framework; }
+  std::shared_ptr<scr::ServiceComponentRuntime> dsRuntimeService;
+  
+private:
+  cppmicroservices::Framework framework;
+};
+
+TEST_F(SharedLibraryExceptionTest, testDSBundleImmediateTrue)
+{
+  auto context = GetFramework().GetBundleContext();
+  test::InstallLib(context, "TestBundleDSSLE1");
+  
+  auto bundles = context.GetBundles();
+  auto bundle = std::find_if( bundles.begin(), bundles.end(), [](const cppmicroservices::Bundle& bundle) {
+    return (bundle.GetSymbolicName() == "TestBundleDSSLE1");
+  });
+  
+  ASSERT_NE(bundle, bundles.end()) << "TestBundleDSSLE1 not found";
+  ASSERT_THROW(bundle->Start(), cppmicroservices::SharedLibraryException);
+}
+
+TEST_F(SharedLibraryExceptionTest, testDSBundleImmediateFalse)
+{
+  auto context = GetFramework().GetBundleContext();
+  test::InstallLib(context, "TestBundleDSSLE2");
+
+  auto bundles = context.GetBundles();
+  auto bundle = std::find_if( bundles.begin(), bundles.end(), [](const cppmicroservices::Bundle& bundle) {
+    return (bundle.GetSymbolicName() == "TestBundleDSSLE2");
+  });
+
+  ASSERT_NE(bundle, bundles.end()) << "TestBundleDSSLE2 not found";
+  bundle->Start();
```

Comment:
In this case `Bundle::Start()` should never throw. I suggest using a ```ASSERT_NO_THROW```

Reply:
made the change accordingly